### PR TITLE
fixed the bug resultTtlInSeconds defaults to 300 when set to 0

### LINKED ISSUE #2587: resultTtlInSeconds defaults to 300 when set to 0
# This is a Bug Report

## Description

* What went wrong?

`resultTtlInSeconds` defaults to 300 when set to 0.

This is oviously due to [this line](https://github.com/serverless/serverless/blob/f817933909cb524dcde729348bf229ae355e15ac/lib/plugins/aws/deploy/compile/events/apiGateway/lib/authorizers.js#L54
) (since 0 is falesy...)

### PR DESCRIPTION
**Pull request by horike37**
## What did you implement:

Closes #2587 

## How did you implement it:
It has returned `false` when set to 0, then fixed to reterun 0.

## How can we verify it:
Please set 0 in `resultTtlInSeconds` statement and run `sls deploy`.

## Todos:

- [x] Write tests
- [ ] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config/commands/resources
- [x] Change ready for review message below

***Is this ready for review?:*** YES

### CODE REVIEW

File: lib/plugins/aws/deploy/compile/events/apiGateway/tests/authorizers.js [hunk 1]
```diff
@@ -117,7 +117,7 @@ describe('#compileAuthorizers()', () => {
       expect(
         awsCompileApigEvents.serverless.service.provider.compiledCloudFormationTemplate
           .Resources.AuthorizerApiGatewayAuthorizer.Properties.AuthorizerResultTtlInSeconds
-      ).to.equal(400);
```
Comment (flomotlik) [inline lib/plugins/aws/deploy/compile/events/apiGateway/tests/authorizers.js]: Could you implement a separate test for testing that 0 gets properly set as well.

**Comment by eahefnawy** [comment]
Beautiful! Thanks @horike37 😊 Looks great! Could you fix the conflicts and i can merge right away

**Comment by horike37** [comment]
@eahefnawy @flomotlik 
Thank you for reviewing!
I fixed the conflicts and the test separated when set to 0.
Please check it.

**Comment by eahefnawy** [comment]
thanks @horike37 😊

**Pull request merged**